[TensorExpr] Remove redundant checks from canHandle in TE fuser.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #42938 [TensorExpr] Cleanup logic in the TensorExpr fuser pass.
* **#42937 [TensorExpr] Remove redundant checks from canHandle in TE fuser.**
* #42936 [TensorExpr] Wrap fuser in a class.

Differential Revision: [D23084408](https://our.internmc.facebook.com/intern/diff/D23084408)

Commit:
[TensorExpr] Remove redundant checks from canHandle in TE fuser.

[ghstack-poisoned]

eellison commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -170,12 +170,6 @@ bool canHandle(Node* node) {
-  if (node->kind() == prim::Loop) {
-    return false; // TODO
-  }
-  if (!allShapesAreKnown(node)) {
```
I think we are no longer checking if the inputs to a node are completely known, just the outputs: https://github.com/pytorch/pytorch/blob/e88d6ccab82252e98f2ef6b3ac09b9c426938632/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L330

So i'm not sure if this is redundant

ZolotukhinM replied:
Hmm, I think you are right. I will keep the check for now (I will probably get rid of it anyway in upcoming changes, but in this PR it's probably still technically needed).

Commit:
Update on "[TensorExpr] Remove redundant checks from canHandle in TE fuser."


Differential Revision: [D23084408](https://our.internmc.facebook.com/intern/diff/D23084408)

[ghstack-poisoned]

Commit:
rebase on "[TensorExpr] Remove redundant checks from canHandle in TE fuser."


Differential Revision: [D23084408](https://our.internmc.facebook.com/intern/diff/D23084408)

[ghstack-poisoned]
